packages/shared/src/contexts/base/BaseSharedUIProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { pick } from 'lodash-unified'
+import { createContext, FC, useContext, useMemo, useState } from 'react'
+import type { SharedComponentOverwrite } from './types'
+
+interface ContextOptions {
+    networkIdentifier: string
+    componentOverwrite: SharedComponentOverwrite
+    updateOverwrite(overwrite: SharedComponentOverwrite): void
+    updateNetworkIdentifier(snsId: string): void
+}
+
+// We can use this from outside of components or hooks
+export const staticSharedUI: ContextOptions = {
+    networkIdentifier: 'unknown',
+    componentOverwrite: {},
+    updateOverwrite: (overwrite: SharedComponentOverwrite) => {
+        staticSharedUI.componentOverwrite = overwrite
+    },
+    updateNetworkIdentifier: (snsId: string) => {
+        staticSharedUI.networkIdentifier = snsId
+    },
+}
+
+const BaseUIContext = createContext<ContextOptions>(staticSharedUI)
+
+export const BaseSharedUIProvider: FC = ({ children }) => {
+    const [overwrite, setOverwrite] = useState<SharedComponentOverwrite>(staticSharedUI.componentOverwrite)
```

Comment:
An SNS adaptor might be destroyed and re-run in dev mode. Please make sure calling `updateOverwrite` and `updateNetworkIdentifier` can make this component refresh, otherwise HMR will be broken.

I suggest to use `ValueRef` + `Subscription` to do this work.

Reply:
Well, `updateOverwrite` and `updateNetworkIdentifier` will be overrode to `setOverwrite` and `setSnsId` since the first run, calling `staticSharedUI.update{Overwrite,NetworkIdentifier}` can make it refresh.

```ts
    const [overwrite, setOverwrite] = useState<SharedComponentOverwrite>(staticSharedUI.componentOverwrite)
    const [snsId, setSnsId] = useState(staticSharedUI.networkIdentifier)

    const contextValue = useMemo(() => {
        const value: ContextOptions = {
            networkIdentifier: snsId,
            componentOverwrite: overwrite,
            updateOverwrite: setOverwrite,
            updateNetworkIdentifier: setSnsId,
        }
        Object.assign(staticSharedUI, pick(value, ['networkIdentifier', 'componentOverwrite']))
        return value
    }, [snsId, overwrite])
```

Reply:
what if there's multiple instance of BaseSharedUIProvider?